### PR TITLE
1.x | webhook: Check whether Privileged is nil before accessing its value

### PR DESCRIPTION
**Pull request by fidencio**
Otherwise we'll just crash.

Fixes: #3111

Signed-off-by: Adel Zaalouk <azaalouk@redhat.com>
Signed-off-by: Fabiano Fidêncio <fidencio@redhat.com>

### CODE REVIEW

**Comment by fidencio** [comment]
The image will have to be rebuilt after this PR is merged.

**Review by liubin — APPROVED** [review]

File: kata-webhook/main.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func annotatePodMutator(ctx context.Context, obj metav1.Object) (bool, error) {
 	}
 
 	for i := range pod.Spec.Containers {
-		if pod.Spec.Containers[i].SecurityContext != nil {
+		if pod.Spec.Containers[i].SecurityContext != nil && pod.Spec.Containers[i].SecurityContext.Privileged != nil {
```
Comment (wainersm) [inline kata-webhook/main.go:53]: hi @fidencio , this change looks good. But, in reason of my ignorance on this subject, allow me to ask one question.

If I understood correctly if this webhook is applied to a non-default namespace then the `Privileged` object *might be* nil, as a result it continues to process (eventually the pod is mutated). My doubt is if the lack of `Privileged` really means the container is not privileged or whether it should be checked another object for non-default namespaces.

Comment (fidencio) [reply kata-webhook/main.go:53]: @wainersm,

I think it does and the reason is here: https://pkg.go.dev/k8s.io/api/core/v1#SecurityContext
There it mentions:
```
	// Run container in privileged mode.
	// Processes in privileged containers are essentially equivalent to root on the host.
	// Defaults to false.
	// +optional
	Privileged *bool `json:"privileged,omitempty" protobuf:"varint,2,opt,name=privileged"`
```
From that I understand that unless it's explicitly set, and set to `true`, it'll be false.

Does this answer your question?

Comment (wainersm) [reply kata-webhook/main.go:53]: @fidencio it does answer! Thanks!

**Review by wainersm — APPROVED** [review]

**Review by amshinde — APPROVED** [review]

**Pull request merged**